### ISSUE TITLE
[Enhancement] Compiled bindings to work when also setting BindingContext in XAML

### ISSUE THREAD

**Issue by JKennedy24** [xamarin/Xamarin.Forms#7698]
## Summary
Not sure if this a bug or a feature request.

When setting the BindingContext in XAML there is then no way to use compiled bindings.

For example

                <custom:MyCustomView **x:DataType="{x:Null}"** BindingContext="{Binding PresenterViewModel}" IsVisible="{Binding IsVisible}" >
                        <Label Text="{Binding Text}"/>
                </custom:MyCustomView>

In the above code the datatype can only be `Null`

If you set it to any object, the two properties will conflict and say that either `PresenterViewModel` or `IsVisible` does not exist.

The bottom line is when ever you set BindingContext in Xaml, there is then no way to set DataType correctly

**Comment by samhouts** [comment]
cc @StephaneDelcroix

**Comment by StephaneDelcroix** [comment]
> Not sure if this a bug or a feature request.

midway between the 2 :)
